In [1]:
import pandas as pd # to read csv file of training and testing dataset

In [10]:
train = pd.read_csv('/content/train_ctrUa4K.csv')
test = pd.read_csv('/content/test_lAUu6dG.csv')
train.head(5)


Loan_ID Gender Married  ... Credit_History Property_Area Loan_Status
0  LP001002   Male      No  ...            1.0         Urban           Y
1  LP001003   Male     Yes  ...            1.0         Rural           N
2  LP001005   Male     Yes  ...            1.0         Urban           Y
3  LP001006   Male     Yes  ...            1.0         Urban           Y
4  LP001008   Male      No  ...            1.0         Urban           Y

[5 rows x 13 columns]

In [11]:
test.head(5)

Loan_ID Gender Married  ... Loan_Amount_Term Credit_History Property_Area
0  LP001015   Male     Yes  ...            360.0            1.0         Urban
1  LP001022   Male     Yes  ...            360.0            1.0         Urban
2  LP001031   Male     Yes  ...            360.0            1.0         Urban
3  LP001035   Male     Yes  ...            360.0            NaN         Urban
4  LP001051   Male      No  ...            360.0            1.0         Urban

[5 rows x 12 columns]

In [12]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


In [13]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            367 non-null    object 
 1   Gender             356 non-null    object 
 2   Married            367 non-null    object 
 3   Dependents         357 non-null    object 
 4   Education          367 non-null    object 
 5   Self_Employed      344 non-null    object 
 6   ApplicantIncome    367 non-null    int64  
 7   CoapplicantIncome  367 non-null    int64  
 8   LoanAmount         362 non-null    float64
 9   Loan_Amount_Term   361 non-null    float64
 10  Credit_History     338 non-null    float64
 11  Property_Area      367 non-null    object 
dtypes: float64(3), int64(2), object(7)
memory usage: 34.5+ KB


**missing value treatment (imputation)**
there are lots of missing value or nan,
we will use mode to fill categorical data and mean to fill numerical data

In [14]:
train['Gender'].fillna(train['Gender'].mode()[0], inplace = True)
train['Married'].fillna(train['Married'].mode()[0], inplace = True)
train['Dependents'].fillna(train['Dependents'].mode()[0], inplace = True)
train['Self_Employed'].fillna(train['Self_Employed'].mode()[0], inplace = True) #categorical imputation

In [15]:
train['LoanAmount'].fillna(train['LoanAmount'].mean(), inplace = True)
train['Loan_Amount_Term'].fillna(train['Loan_Amount_Term'].mean(), inplace = True)
train['Credit_History'].fillna(train['Credit_History'].mean(), inplace = True) #numeric imputation

In [17]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             614 non-null    object 
 2   Married            614 non-null    object 
 3   Dependents         614 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      614 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         614 non-null    float64
 9   Loan_Amount_Term   614 non-null    float64
 10  Credit_History     614 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


In [9]:
train_df['Loan_Status'].value_counts()

Y    422
N    192
Name: Loan_Status, dtype: int64

**dataset is imbalanced**
yes is 422 and no is 192
yes--> is 68%
no is--> 32%
we have to balance it using oversampling by synthetic Synthetic Minority Oversampling Technique

test data imputation

In [18]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            367 non-null    object 
 1   Gender             356 non-null    object 
 2   Married            367 non-null    object 
 3   Dependents         357 non-null    object 
 4   Education          367 non-null    object 
 5   Self_Employed      344 non-null    object 
 6   ApplicantIncome    367 non-null    int64  
 7   CoapplicantIncome  367 non-null    int64  
 8   LoanAmount         362 non-null    float64
 9   Loan_Amount_Term   361 non-null    float64
 10  Credit_History     338 non-null    float64
 11  Property_Area      367 non-null    object 
dtypes: float64(3), int64(2), object(7)
memory usage: 34.5+ KB


In [19]:
test.isnull().sum()

Loan_ID               0
Gender               11
Married               0
Dependents           10
Education             0
Self_Employed        23
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount            5
Loan_Amount_Term      6
Credit_History       29
Property_Area         0
dtype: int64

In [20]:
test['Gender'].fillna(test['Gender'].mode()[0], inplace = True)
test['Dependents'].fillna(test['Dependents'].mode()[0], inplace = True)
test['Self_Employed'].fillna(test['Self_Employed'].mode()[0], inplace = True)
test['LoanAmount'].fillna(test['LoanAmount'].mean(), inplace = True)
test['Loan_Amount_Term'].fillna(test['Loan_Amount_Term'].mean(), inplace = True)
test['Credit_History'].fillna(test['Credit_History'].mean(), inplace = True)

In [21]:
test.isnull().sum()

Loan_ID              0
Gender               0
Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
dtype: int64

In [22]:
train_df=train.drop(['Loan_ID'],axis=1)
test_df=test.drop(['Loan_ID'],axis=1)
train_df

Gender Married Dependents  ... Credit_History Property_Area  Loan_Status
0      Male      No          0  ...            1.0         Urban            Y
1      Male     Yes          1  ...            1.0         Rural            N
2      Male     Yes          0  ...            1.0         Urban            Y
3      Male     Yes          0  ...            1.0         Urban            Y
4      Male      No          0  ...            1.0         Urban            Y
..      ...     ...        ...  ...            ...           ...          ...
609  Female      No          0  ...            1.0         Rural            Y
610    Male     Yes         3+  ...            1.0         Rural            Y
611    Male     Yes          1  ...            1.0         Urban            Y
612    Male     Yes          2  ...            1.0         Urban            Y
613  Female      No          0  ...            0.0     Semiurban            N

[614 rows x 12 columns]

**one hot encoding**
to convert categorical variables into numeric for ml models to run

In [23]:
train_one_hot= pd.get_dummies(train_df)

In [24]:
train_one_hot

ApplicantIncome  CoapplicantIncome  ...  Loan_Status_N  Loan_Status_Y
0               5849                0.0  ...              0              1
1               4583             1508.0  ...              1              0
2               3000                0.0  ...              0              1
3               2583             2358.0  ...              0              1
4               6000                0.0  ...              0              1
..               ...                ...  ...            ...            ...
609             2900                0.0  ...              0              1
610             4106                0.0  ...              0              1
611             8072              240.0  ...              0              1
612             7583                0.0  ...              0              1
613             4583                0.0  ...              1              0

[614 rows x 22 columns]

In [25]:
test_one_hot= pd.get_dummies(test_df)

In [26]:
test_one_hot

ApplicantIncome  ...  Property_Area_Urban
0               5720  ...                    1
1               3076  ...                    1
2               5000  ...                    1
3               2340  ...                    1
4               3276  ...                    1
..               ...  ...                  ...
362             4009  ...                    1
363             4158  ...                    1
364             3250  ...                    0
365             5000  ...                    0
366             9200  ...                    0

[367 rows x 20 columns]

In [31]:
X= train_one_hot.drop(['Loan_Status_N','Loan_Status_Y'],axis=1)
y = train_one_hot['Loan_Status_Y']
X

ApplicantIncome  ...  Property_Area_Urban
0               5849  ...                    1
1               4583  ...                    0
2               3000  ...                    1
3               2583  ...                    1
4               6000  ...                    1
..               ...  ...                  ...
609             2900  ...                    0
610             4106  ...                    0
611             8072  ...                    1
612             7583  ...                    1
613             4583  ...                    0

[614 rows x 20 columns]

In [32]:
y

0      1
1      0
2      1
3      1
4      1
      ..
609    1
610    1
611    1
612    1
613    0
Name: Loan_Status_Y, Length: 614, dtype: uint8

**train test split**

In [34]:
from sklearn.model_selection import train_test_split
X_train,X_valid,y_train,y_valid = train_test_split(X,y,test_size=0.1,random_state=1)

baseline model using decision tree without balancing dataset

In [35]:
from sklearn.tree import DecisionTreeClassifier
base_model = DecisionTreeClassifier()
base_model.fit(X_train,y_train)
y_pred_1 = base_model.predict(X_valid)
from sklearn.metrics import accuracy_score
acc = accuracy_score(y_valid,y_pred_1)
acc

0.6935483870967742

**balancing dataset using SMOTE**

In [36]:
from imblearn.over_sampling import SMOTE
oversample=SMOTE(k_neighbors=2)
#transform the dataset
X,y=oversample.fit_resample(X,y)

/usr/local/lib/python3.7/dist-packages/sklearn/externals/six.py:31: FutureWarning: The module is deprecated in version 0.21 and will be removed in version 0.23 since we've dropped support for Python 2.7. Please rely on the official version of six (https://pypi.org/project/six/).
  "(https://pypi.org/project/six/).", FutureWarning)
/usr/local/lib/python3.7/dist-packages/sklearn/utils/deprecation.py:144: FutureWarning: The sklearn.neighbors.base module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.neighbors. Anything that cannot be imported from sklearn.neighbors is now part of the private API.
  warnings.warn(message, FutureWarning)
/usr/local/lib/python3.7/dist-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function safe_indexing is deprecated; safe_indexing is deprecated in version 0.22 and will be removed in version 0.24.
  warnings.warn(msg, category=FutureWarning)


In [43]:
X_bal=pd.DataFrame(X)
X_bal

0            1           2   ...        17        18        19
0     5849.000000     0.000000  146.412162  ...  0.000000  0.000000  1.000000
1     4583.000000  1508.000000  128.000000  ...  1.000000  0.000000  0.000000
2     3000.000000     0.000000   66.000000  ...  0.000000  0.000000  1.000000
3     2583.000000  2358.000000  120.000000  ...  0.000000  0.000000  1.000000
4     6000.000000     0.000000  141.000000  ...  0.000000  0.000000  1.000000
..            ...          ...         ...  ...       ...       ...       ...
839   4697.510234     0.000000  110.100640  ...  0.871855  0.000000  0.128145
840   2389.020426     0.000000   51.606883  ...  0.806659  0.000000  0.193341
841   2389.011783     0.000000   51.602486  ...  0.806811  0.000000  0.193189
842  14905.860172   618.967211  260.274270  ...  0.294746  0.705254  0.000000
843   4395.351847     0.000000  127.418334  ...  0.000000  1.000000  0.000000

[844 rows x 20 columns]

In [44]:
y_bal=pd.DataFrame(y)
y_bal.value_counts()

1    422
0    422
dtype: int64

In [45]:
X_train_bal,X_valid_bal,y_train_bal,y_valid_bal = train_test_split(X_bal,y_bal,test_size=0.1,random_state=1)

**model after balancing** 

In [47]:
model_1 = DecisionTreeClassifier()
model_1.fit(X_train_bal,y_train_bal)
y_pred_1 = model_1.predict(X_valid_bal)
acc_1 = accuracy_score(y_valid_bal,y_pred_1)
acc_1

0.8117647058823529

**normalization of data for better performance**

In [48]:
from sklearn.preprocessing import StandardScaler
scaler=StandardScaler()
X_train_bal_norm = scaler.fit_transform(X_train_bal)
X_valid_bal_norm =scaler.fit_transform(X_valid_bal)

In [49]:
X_train_bal_norm

array([[ 0.14848284, -0.48594519,  0.76654856, ..., -0.14639535,
         0.89294163, -0.7556019 ],
       [-0.10469857,  0.34487591, -0.06098949, ..., -0.71664112,
         1.44232618, -0.7556019 ],
       [-0.58597827,  0.99259513, -0.94585799, ...,  1.55015126,
        -0.74154047, -0.7556019 ],
       ...,
       [-0.30278843, -0.48594519, -0.77615718, ..., -0.71664112,
         1.44232618, -0.7556019 ],
       [ 0.0547745 , -0.11798927,  0.30265509, ...,  1.55015126,
        -0.74154047, -0.7556019 ],
       [-0.27293193, -0.0598756 , -0.01250355, ..., -0.71664112,
         1.44232618, -0.7556019 ]])

In [50]:
model_2 = DecisionTreeClassifier()
model_2.fit(X_train_bal_norm,y_train_bal)
y_pred_2 =model_2.predict(X_valid_bal_norm)
acc_2 = accuracy_score(y_valid_bal,y_pred_2)
acc_2

0.5764705882352941

**random forest classifier for model_3**

In [51]:
from sklearn.ensemble import RandomForestClassifier
model_3 = RandomForestClassifier()
model_3.fit(X_train_bal,y_train_bal)
y_pred_3 = model_3.predict(X_valid_bal)
acc_3 = accuracy_score(y_valid_bal,y_pred_3)
acc_3



/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  This is separate from the ipykernel package so we can avoid doing imports until


0.8705882352941177

In [52]:
from sklearn.ensemble import RandomForestClassifier
model_4 = RandomForestClassifier()
model_4.fit(X_train_bal_norm,y_train_bal)
y_pred_4 = model_3.predict(X_valid_bal_norm)
acc_4 = accuracy_score(y_valid_bal,y_pred_4)
acc_4

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  This is separate from the ipykernel package so we can avoid doing imports until


0.5411764705882353

**EXTRATREE CLASSIFIER**

In [54]:
from sklearn.ensemble import ExtraTreesClassifier
model_5 = ExtraTreesClassifier()
model_5.fit(X_train_bal,y_train_bal)
y_pred_5 = model_5.predict(X_valid_bal)
acc_5 = accuracy_score(y_valid_bal,y_pred_5)
acc_5

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  This is separate from the ipykernel package so we can avoid doing imports until


0.8

**LOGISTIC REGRESSION** *italicized text*

In [55]:
from sklearn.linear_model import LogisticRegression
model_6 = LogisticRegression()
model_6.fit(X_train_bal,y_train_bal)
y_pred_6 = model_6.predict(X_valid_bal)
acc_6 = accuracy_score(y_valid_bal,y_pred_6)
acc_6

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


0.7176470588235294

**SUPPORT VECTOR MACHINE**

In [56]:
from sklearn.svm import SVC
model_7 = SVC()
model_7.fit(X_train_bal,y_train_bal)
y_pred_7 = model_7.predict(X_valid_bal)
acc_7 = accuracy_score(y_valid_bal,y_pred_7)
acc_7

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


0.49411764705882355

**BEST ML MODEL WAS RANDOM FOREST**
WITH ACCURACY OF 87%

**HYPERPARAMETER TUNING OF RANDOM FOREST**

In [76]:
from sklearn.ensemble import RandomForestClassifier
model_3 = RandomForestClassifier(n_estimators=200,random_state=1)
model_3.fit(X_train_bal,y_train_bal)
y_pred_3 = model_3.predict(X_valid_bal)
acc_3 = accuracy_score(y_valid_bal,y_pred_3)
acc_3


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  This is separate from the ipykernel package so we can avoid doing imports until


0.8588235294117647

**DEEP LEARNING MODEL USING TENSORFLOW**

In [89]:
import tensorflow as tf
from tensorflow import keras
from keras import layers

In [103]:
tf.random.set_seed(1)
model_8 = keras.models.Sequential([
                                   keras.layers.Dense(50,activation='relu'),
                                   keras.layers.Dense(40,activation='relu'),
                                   keras.layers.Dense(1,activation='sigmoid')
])


In [104]:
model_8.compile(optimizer='Adam',loss=keras.losses.CategoricalHinge(),
                metrics=['accuracy'])

In [108]:
model_8.fit(X_train_bal_norm,y_train_bal,epochs=100)

Epoch 1/100
24/24 [==============================] - 0s 2ms/step - loss: 0.6320 - accuracy: 0.8722
Epoch 2/100
24/24 [==============================] - 0s 3ms/step - loss: 0.6304 - accuracy: 0.8709
Epoch 3/100
24/24 [==============================] - 0s 2ms/step - loss: 0.6295 - accuracy: 0.8762
Epoch 4/100
24/24 [==============================] - 0s 2ms/step - loss: 0.6285 - accuracy: 0.8801
Epoch 5/100
24/24 [==============================] - 0s 3ms/step - loss: 0.6275 - accuracy: 0.8775
Epoch 6/100
24/24 [==============================] - 0s 2ms/step - loss: 0.6286 - accuracy: 0.8748
Epoch 7/100
24/24 [==============================] - 0s 2ms/step - loss: 0.6272 - accuracy: 0.8762
Epoch 8/100
24/24 [==============================] - 0s 3ms/step - loss: 0.6261 - accuracy: 0.8788
Epoch 9/100
24/24 [==============================] - 0s 2ms/step - loss: 0.6254 - accuracy: 0.8801
Epoch 10/100
24/24 [==============================] - 0s 2ms/step - loss: 0.6249 - accuracy: 0.8827
Epoch 11/

In [109]:
y_pred_8 = model_8.predict(X_valid_bal_norm)
model_8.evaluate(X_valid_bal_norm,y_valid_bal)

3/3 [==============================] - 0s 4ms/step - loss: 0.7971 - accuracy: 0.7529


[0.797107458114624, 0.7529411911964417]

**TIME TO GET THE PREDICTION USING RANDOM FOREST AND DEEP LEARNING MODEL FOR TEST DATASET**

In [110]:
test_one_hot

ApplicantIncome  ...  Property_Area_Urban
0               5720  ...                    1
1               3076  ...                    1
2               5000  ...                    1
3               2340  ...                    1
4               3276  ...                    1
..               ...  ...                  ...
362             4009  ...                    1
363             4158  ...                    1
364             3250  ...                    0
365             5000  ...                    0
366             9200  ...                    0

[367 rows x 20 columns]

In [112]:
prediction_by_RandomForest = model_3.predict(test_one_hot)
prediction_by_RandomForest

array([1, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0,
       1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1,
       0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0,
       1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0,
       1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1,
       1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0,
       1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1,
       1, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1,

In [115]:
prediction_by_deeplearning = model_8.predict(test_one_hot)
prediction_by_deeplearning

array([[1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [0.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],
       [1.],

In [116]:
sub= pd.read_csv('/content/sample_submission_49d68Cx.csv')
sub

Loan_ID Loan_Status
0    LP001015           N
1    LP001022           N
2    LP001031           N
3    LP001035           N
4    LP001051           N
..        ...         ...
362  LP002971           N
363  LP002975           N
364  LP002980           N
365  LP002986           N
366  LP002989           N

[367 rows x 2 columns]

In [118]:
test['Loan_Status']=prediction_by_RandomForest

In [119]:
test

Loan_ID Gender Married  ... Credit_History Property_Area Loan_Status
0    LP001015   Male     Yes  ...       1.000000         Urban           1
1    LP001022   Male     Yes  ...       1.000000         Urban           1
2    LP001031   Male     Yes  ...       1.000000         Urban           1
3    LP001035   Male     Yes  ...       0.825444         Urban           1
4    LP001051   Male      No  ...       1.000000         Urban           0
..        ...    ...     ...  ...            ...           ...         ...
362  LP002971   Male     Yes  ...       1.000000         Urban           1
363  LP002975   Male     Yes  ...       1.000000         Urban           1
364  LP002980   Male      No  ...       0.825444     Semiurban           0
365  LP002986   Male     Yes  ...       1.000000         Rural           1
366  LP002989   Male      No  ...       1.000000         Rural           1

[367 rows x 13 columns]

In [122]:
pr=pd.DataFrame(prediction_by_RandomForest)
pr

0
0    1
1    1
2    1
3    1
4    0
..  ..
362  1
363  1
364  0
365  1
366  1

[367 rows x 1 columns]

In [130]:
test.Loan_Status.replace((1, 0), ('Y', 'N'), inplace=True)

In [131]:
test

Loan_ID Gender Married  ... Credit_History Property_Area Loan_Status
0    LP001015   Male     Yes  ...       1.000000         Urban           Y
1    LP001022   Male     Yes  ...       1.000000         Urban           Y
2    LP001031   Male     Yes  ...       1.000000         Urban           Y
3    LP001035   Male     Yes  ...       0.825444         Urban           Y
4    LP001051   Male      No  ...       1.000000         Urban           N
..        ...    ...     ...  ...            ...           ...         ...
362  LP002971   Male     Yes  ...       1.000000         Urban           Y
363  LP002975   Male     Yes  ...       1.000000         Urban           Y
364  LP002980   Male      No  ...       0.825444     Semiurban           N
365  LP002986   Male     Yes  ...       1.000000         Rural           Y
366  LP002989   Male      No  ...       1.000000         Rural           Y

[367 rows x 13 columns]

In [135]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            367 non-null    object 
 1   Gender             367 non-null    object 
 2   Married            367 non-null    object 
 3   Dependents         367 non-null    object 
 4   Education          367 non-null    object 
 5   Self_Employed      367 non-null    object 
 6   ApplicantIncome    367 non-null    int64  
 7   CoapplicantIncome  367 non-null    int64  
 8   LoanAmount         367 non-null    float64
 9   Loan_Amount_Term   367 non-null    float64
 10  Credit_History     367 non-null    float64
 11  Property_Area      367 non-null    object 
 12  Loan_Status        367 non-null    object 
dtypes: float64(3), int64(2), object(8)
memory usage: 37.4+ KB


In [136]:
sub_df=pd.DataFrame({'Loan_ID' : test["Loan_ID"],'Loan_Status' : test['Loan_Status']})

In [137]:
sub_df

Loan_ID Loan_Status
0    LP001015           Y
1    LP001022           Y
2    LP001031           Y
3    LP001035           Y
4    LP001051           N
..        ...         ...
362  LP002971           Y
363  LP002975           Y
364  LP002980           N
365  LP002986           Y
366  LP002989           Y

[367 rows x 2 columns]

In [138]:
submission_ran_for = sub_df.to_csv('submission_random_forest.csv',index=False)

In [139]:
pr1=pd.DataFrame(prediction_by_deeplearning)
pr1

0
0    1.0
1    1.0
2    1.0
3    1.0
4    1.0
..   ...
362  1.0
363  1.0
364  1.0
365  1.0
366  1.0

[367 rows x 1 columns]

In [140]:
test['Loan_Status']=pr1


In [141]:
test

Loan_ID Gender Married  ... Credit_History Property_Area Loan_Status
0    LP001015   Male     Yes  ...       1.000000         Urban         1.0
1    LP001022   Male     Yes  ...       1.000000         Urban         1.0
2    LP001031   Male     Yes  ...       1.000000         Urban         1.0
3    LP001035   Male     Yes  ...       0.825444         Urban         1.0
4    LP001051   Male      No  ...       1.000000         Urban         1.0
..        ...    ...     ...  ...            ...           ...         ...
362  LP002971   Male     Yes  ...       1.000000         Urban         1.0
363  LP002975   Male     Yes  ...       1.000000         Urban         1.0
364  LP002980   Male      No  ...       0.825444     Semiurban         1.0
365  LP002986   Male     Yes  ...       1.000000         Rural         1.0
366  LP002989   Male      No  ...       1.000000         Rural         1.0

[367 rows x 13 columns]

In [142]:
test.Loan_Status.replace((1, 0), ('Y', 'N'), inplace=True)

In [144]:
test.head(3)

Loan_ID Gender Married  ... Credit_History Property_Area Loan_Status
0  LP001015   Male     Yes  ...            1.0         Urban           Y
1  LP001022   Male     Yes  ...            1.0         Urban           Y
2  LP001031   Male     Yes  ...            1.0         Urban           Y

[3 rows x 13 columns]

In [146]:
sub_df1=pd.DataFrame({'Loan_ID' : test["Loan_ID"],'Loan_Status' : test['Loan_Status']})
sub_df1

Loan_ID Loan_Status
0    LP001015           Y
1    LP001022           Y
2    LP001031           Y
3    LP001035           Y
4    LP001051           Y
..        ...         ...
362  LP002971           Y
363  LP002975           Y
364  LP002980           Y
365  LP002986           Y
366  LP002989           Y

[367 rows x 2 columns]

In [147]:
sub_df1=pd.DataFrame({'Loan_ID' : test["Loan_ID"],'Loan_Status' : test['Loan_Status']})
submission_deeplearning = sub_df1.to_csv('submission_deep_learning.csv',index=False)